mitiq/interface/mitiq_pennylane/conversions.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Functions to convert between Mitiq's internal circuit representation and
+Pennylane's circuit representation.
+"""
+
+from cirq import Circuit
+from mitiq.interface.mitiq_qiskit import from_qasm as cirq_from_qasm, to_qasm
+from pennylane.wires import Wires
+from pennylane.tape import QuantumTape
+
+from pennylane import from_qasm as pennylane_from_qasm
+
+
+def from_pennylane(tape: QuantumTape) -> Circuit:
+    """Returns a Cirq circuit equivalent to the input QuantumTape.
+
+    Args:
+        tape: Pennylane QuantumTape to convert to a Cirq circuit.
+    """
+    return cirq_from_qasm(tape.to_openqasm(rotations=True))
```

Comment:
It's a good point, thanks for bringing this up! Without regard to folding and just thinking about `from_pennylane`, it should include the rotations, otherwise the Cirq circuit - which only supports computational basis measurements - will not be equivalent.

However this makes it so that the Pennylane -> Cirq -> Pennylane sequence can start with non-computational basis measurements and end with computational basis measurements, which is a little strange. With hindsight we've realized conversions typically have problems like this, and have handled it in a "first do sloppy conversion, then clean it up" way. (Two examples of cleaning it up are [adding back PyQuil declarations](https://github.com/unitaryfund/mitiq/blob/9140ec2080651fe621f65d61d1ab2fb13e4ca680/mitiq/interface/conversions.py#L230-L265) and [keeping the same register structure in Qiskit](https://github.com/unitaryfund/mitiq/blob/9140ec2080651fe621f65d61d1ab2fb13e4ca680/mitiq/interface/conversions.py#L267-L282).) This is a con of the design in my opinion, but the pro is that as long as you have `to_whatever` and `from_whatever`, even if it's (necessarily) a little sloppy, it makes it very easy to add support for `whatever`.

Anyway, you asked about folding the rotations: Since all other platforms measure in the Z basis we do fold rotations by default. There are options to not fold single-qubit gates, or certain types of gates, so whether or not rotations are there doesn't matter too much from the perspective of folding.

Reply:
What about having the `rotations` argument in the function `from_pennylane`?

It seems that if we consider `from_pennylane` as a stand-alone function, the natural choice would be `rotations=True` .

On the other hand, if we also want to restore the full metadata of the tape (return type, observable, etc.) somewhere along the Mitiq worflow, it is probably easier if one can set `rotations=False` to decouple the bare circuit from the rest.